packages/core/addon/models/table.js

```diff
@@ -77,22 +78,63 @@ export default VisualizationBase.extend(Validations, {
   }
 });
 
+/**
+ * returns default dimension fields (show clause)
+ * @param {Object} dimension - dimension from request
+ * @returns {Object} - list of field names
+ */
+function getDefaultDimensionFields(dimension) {
+  let fields = get(dimension, 'dimension.fields');
+
+  if (isArray(fields)) {
+    return fields.filter(field => isArray(field.tags) && field.tags.includes('show')).map(field => field.name);
+  }
+
+  return [];
+}
+
+/**
+ * builds a single dimension column
+ * @param {Object} dimension - dimension from request
+ * @param {Object} column
+ * @param {String} fieldName
+ * @returns {Object} - dimension column
+ */
+function dimensionColumn(dimension, column, fieldName) {
```

Comment:
should we call this `buildDimensionColumn`?

Reply:
yeah tried to avoid having `buildDimensionColumns` and `buildDimensionColumn` but I also prefer `buildDimensionColumns`